deepinv/optim/fixed_point.py

```diff
@@ -101,13 +109,101 @@ def __init__(
         self.update_metrics_fn = update_metrics_fn
         self.check_conv_fn = check_conv_fn
         self.check_iteration_fn = check_iteration_fn
+        self.anderson_acceleration = anderson_acceleration
+        self.history_size = history_size
+        self.beta_anderson_acc = beta_anderson_acc
+        self.eps_anderson_acc = eps_anderson_acc
 
         if self.check_conv_fn is None and self.early_stop:
             warnings.warn(
                 "early_stop is set to True but no check_conv_fn has been defined."
             )
             self.early_stop = False
 
+    def init_anderson_acceleration(self, X):
+        r"""
+        Initialize the Anderson acceleration algorithm.
+
+        :param dict X: initial iterate.
+        """
+        x = X["est"][0]
+        b, d, h, w = x.shape
+        x_hist = torch.zeros(
+            b, self.history_size, d * h * w, dtype=x.dtype, device=x.device
+        )  # history of iterates.
+        T_hist = torch.zeros(
+            b, self.history_size, d * h * w, dtype=x.dtype, device=x.device
+        )  # history of T(x_k) with T the fixed point operator.
+        H = torch.zeros(
+            b,
+            self.history_size + 1,
+            self.history_size + 1,
+            dtype=x.dtype,
+            device=x.device,
+        )  # H in the Anderson acceleration linear system Hp = q .
+        H[:, 0, 1:] = H[:, 1:, 0] = 1.0
+        q = torch.zeros(
+            b, self.history_size + 1, 1, dtype=x.dtype, device=x.device
+        )  # q in the Anderson acceleration linear system Hp = q .
+        q[:, 0] = 1
+        return x_hist, T_hist, H, q
+
+    def anderson_acceleration_step(
+        self,
+        it,
+        X_prev,
+        x_hist,
+        T_hist,
+        H,
+        q,
+        cur_data_fidelity,
+        cur_prior,
+        cur_params,
+        *args
+    ):
+        r"""
+        Anderson acceleration step.
+
+        :param int it: current iteration.
+        :param dict X_prev: previous iterate.
+        :param torch.Tensor x_hist: history of last ``history-size`` iterates.
+        :param torch.Tensor T_hist: history of T evlauaton at the last ``history-size``, where T is the fixed-point operator.
+        :param torch.Tensor H: H in the Anderson acceleration linear system Hp = q .
+        :param torch.Tensor q: q in the Anderson acceleration linear system Hp = q .
+        :param deepinv.optim.DataFidelity cur_data_fidelity: Instance of the DataFidelity class defining the current data_fidelity.
+        :param deepinv.optim.prior cur_prior: Instance of the Prior class defining the current prior.
+        :param dict cur_params: Dictionary containing the current parameters of the algorithm.
+        :param args: arguments for the iterator.
+        """
+        TX = self.iterator(X_prev, cur_data_fidelity, cur_prior, cur_params, *args)
```

Comment:
Maybe take this out of the function? Like that anderson is the same thing as regular but it's just an additional computation after the `self.iterator` call. Seems possible to me?

Reply:
sounds good